docs/certificate_management.md

```diff
@@ -0,0 +1,226 @@
+
+**SDS (Secret Discovery Service) response** *(Diagram object 2)* - When the ADS server is created it will pass an SDS handler. This SDS handler is used to create an SDS response to the proxies requesting secrets. Part of the secret response is the certificate.
+
+**Ingress gateway certificate** *(Diagram object 3)* - A certificate will be issued to the ingress controller when ingress is initialized. A goroutine will be started to monitor for different changes (mesh config change and cert rotation). Certificate rotation will also make a call to `IssueCertificate`.
+
+**Validating webhook** *(Diagram object 4)* - When OSM controller starts up it will create a validating webhook server. A certificate will be issued for this webhook server. This certificate is stored in the OSM control plane namespace and named `validating-webhook-cert-secret`.
+
+**Conversion webhook** *(Diagram object 5)* - When OSM bootstrap starts it will create a web server for the CRD conversion webhook. A certificate will be issued for this web server. This certificate is stored in the OSM control plane namespace and named `crd-converter-cert-secret`.
+
+**Mutating webhook** *(Diagram object 6)* - When OSM injector starts it will create a web server for the mutating webhook to inject the proxy into the pods joined to the mesh. This web server will have a certificate issued for it. This certificate is stored in the OSM control plane namespace and named `mutating-webhook-cert-secret`.
+
+**Mutating webhook patch** *(Diagram object 7)* - Part of the mutating webhook implementation is to create a patch to inject the proxy sidecar. A bootstrap certificate will be issued so that the Envoy sidecar can communicate with the XDS server. The bootstrap config (including this certificate) will be stored in the same namespace as the pod as a secret with the name `envoy-bootstrap-config-<proxy_id>`.
+
+![OSM certificate management diagram](./images/osm_certificate_management1.png)
+
+## Certificate rotation
+
+Each of the certificate managers will run a goroutine to that will check certificate expiration (currently this is hardcoded to every 5 seconds). This goroutine will loop through all certificates from the certificate manager and check to see if the certificates are within 30 seconds of expiration (with additional noise factored in). If so, the certificate will be rotated.
+
+## Root certificate
+
+The root certificate is stored by default in the OSM control plane namespace and named `osm-ca-bundle` when using the builtin certificate manager (tresor). The root certificate is what is used for the certificate manager to issue certificates. For example, the metadata for the root certificate in an installation:
+
+```
+$ kubectl get secret -n osm-system osm-ca-bundle -o jsonpath='{.data.ca\.crt}' | base64 -d | openssl x509 -noout -text
```

Comment:
Per https://github.com/openservicemesh/osm-docs/issues/87 and to align with osm docs, it may make sense to change all references to `osm-system` to an exported environment variable as done in https://github.com/openservicemesh/osm-docs/pull/225 ?

Reply:
I made a note that if you're targeting a different OSM namespace then to substitute `osm-system` with that namespace. In the event of having a small article like this on internals I think that's ok to keep `osm-system` in there with that note. There are a lot of sample commands that include their outputs, which have `osm-system` as the namespace so this seems like a good approach.